src/extras/module-types.js

```diff
@@ -33,6 +33,9 @@
       if (cssContentType.test(contentType))
         return res.text()
         .then(function (source) {
+          source = source.replace(/url\(\s*?(["']??)((?:\\.|[^\s:"'])+?)\1\s*?\)/g, function (match, quotes, id) {
```

Comment:
Strictly speaking, the quotes should be expanded and matched here - `"..."` OR `'...'`. This would currently match eg `"..'.."` incorrectly.

Reply:
Sorry, I can't reproduce this problem.

Below are three edge cases that won't be matched.
>[https://regex101.com/r/uQkG3W/4/](https://regex101.com/r/uQkG3W/4/)
><img width="678" alt="Screen Shot 2021-04-30 at 22 50 57" src="https://user-images.githubusercontent.com/2867188/116712665-aa61e480-aa06-11eb-8fe4-513b6b31cc23.png">

And the replaced source will keep original single quote or double quote.
> matches the same quote as the first capturing group
><img width="543" alt="Screen Shot 2021-04-30 at 22 54 12" src="https://user-images.githubusercontent.com/2867188/116713274-3411b200-aa07-11eb-865b-2e2bdf00a6e3.png">

> before replaced
><img width="724" alt="Screen Shot 2021-04-30 at 22 46 00" src="https://user-images.githubusercontent.com/2867188/116713341-45f35500-aa07-11eb-9a38-1184aa013056.png">
> after replaced
><img width="732" alt="Screen Shot 2021-04-30 at 22 46 54" src="https://user-images.githubusercontent.com/2867188/116713812-b8643500-aa07-11eb-9a6a-f7be50a0f1fa.png">

Reply:
Ah thanks for the clarification. I did some experimenting and would suggest using:

```
url\(\s*(["'])?(\\.|[^"')\\])+\1\s*\)
```

This more flexibly handles string escapes I think.

Reply:
Well, we can't match relative url with this RegEx.
```
url\(\s*(["'])?(\\.|[^"')\\])+\1\s*\)
```
<img width="695" alt="Screen Shot 2021-04-30 at 23 26 57" src="https://user-images.githubusercontent.com/2867188/116717767-d59b0280-aa0b-11eb-87c2-9a3984ef051e.png">

Do you mean this? But it also matches blob and absolute urls, which I think it is unnecessary to normalize.
```
url\(\s*(["'])?((?:\\.|[^"')\\])+)\1\s*\)
```
<img width="694" alt="Screen Shot 2021-04-30 at 23 31 57" src="https://user-images.githubusercontent.com/2867188/116718211-4fcb8700-aa0c-11eb-842b-a7c386687f2d.png">

Reply:
Both `images/myImg.jpg` and `./images/myImg.jpg` are partial URLs to resolve relative to the stylesheet though right?

Reply:
Yes, you are right.
<img width="693" alt="Screen Shot 2021-05-01 at 00 03 43" src="https://user-images.githubusercontent.com/2867188/116722578-28c38400-aa11-11eb-8287-92674a894ad4.png">

Reply:
The normalization of those will be ignored anyway through URL detection, so I think it's ok.

Reply:
The regex that you suggested matches single character `.` `g` `h` rather than partial url like `images/myImg.jpg`.
```
url\(\s*(["'])?(\\.|[^"')\\])+\1\s*\)
```
<img width="695" alt="Screen Shot 2021-04-30 at 23 26 57" src="https://user-images.githubusercontent.com/2867188/116717767-d59b0280-aa0b-11eb-87c2-9a3984ef051e.png">

Reply:
Ah right, yes `url\(\s*(["'])?((?:\\.|[^"')\\])+)\1\s*\)` as the intention exactly.

Reply:
Updated, please take a look.

Reply:
Some browsers like IE might autocorrect invalid URLs in double quotes with unencoded whitespace characters in it, but we shouldn’t rely on this. Keep invalid URLs untouched and normalize the validated is a good choice as far as I think.